### ISSUE TITLE
create new document with relationship

### ISSUE THREAD

**Issue by sajadghawami** [graphql-compose/graphql-compose-elasticsearch#102]
hey there,

first of all, thank you very much for this awsome library! :)

So what i am trying to do is, i am saving a document with mongoose, which works fine so far. The only problem is a relationship that i have. When removing this relationship, everything works fine.

Thats the response:

```
node_1           | Elasticsearch TRACE: 2020-05-06T07:24:46Z
node_1           |   -> POST http://elasticsearch:9200/sellerproduct/_doc/5eb2663ee8944102a2f62e6a
node_1           |   {
node_1           |     "title": "sometitle123",
node_1           |     "category": "category123asfasf",
node_1           |     "price": 1,
node_1           |     "shortDescription": "short desc123rasda",
node_1           |     "longDescription": "long descr",
node_1           |     "seller": {
node_1           |       "_id": "5eb19ff8044ab800928c613b",
node_1           |       "email": "seller1",
node_1           |       "password": "dc13c1846086221cd4fea00f71ddefb03f3b34b364e469365358fa1348e647504866c0ac05bdeefa53d193e3b3ef8747a9d3658fab5b8deb246400880004e9ad",
node_1           |       "salt": "ac206ab5b82ad552",
node_1           |       "createdAt": "2020-05-05T17:18:48.093Z",
node_1           |       "updatedAt": "2020-05-05T17:18:48.097Z",
node_1           |       "__v": 0,
node_1           |       "sellerAddress": "5eb19ff8044ab800928c613c"
node_1           |     },
node_1           |     "image": "5eb2663ee8944102a2f62e69",
node_1           |     "updatedAt": "2020-05-06T07:24:46.151Z",
node_1           |     "createdAt": "2020-05-06T07:24:46.151Z"
node_1           |   }
node_1           |   <- 400
node_1           |   {
node_1           |     "error": {
node_1           |       "root_cause": [
node_1           |         {
node_1           |           "type": "mapper_parsing_exception",
node_1           |           "reason": "failed to parse field [seller] of type [text] in document with id \u00275eb2663ee8944102a2f62e6a\u0027. Preview of field\u0027s value: \u0027{createdAt=2020-05-05T17:18:48.093Z, password=dc13c1846086221cd4fea00f71ddefb03f3b34b364e469365358fa1348e647504866c0ac05bdeefa53d193e3b3ef8747a9d3658fab5b8deb246400880004e9ad, salt=ac206ab5b82ad552, __v=0, sellerAddress=5eb19ff8044ab800928c613c, _id=5eb19ff8044ab800928c613b, email=seller1, updatedAt=2020-05-05T17:18:48.097Z}\u0027"
node_1           |         }
node_1           |       ],
node_1           |       "type": "mapper_parsing_exception",
node_1           |       "reason": "failed to parse field [seller] of type [text] in document with id \u00275eb2663ee8944102a2f62e6a\u0027. Preview of field\u0027s value: \u0027{createdAt=2020-05-05T17:18:48.093Z, password=dc13c1846086221cd4fea00f71ddefb03f3b34b364e469365358fa1348e647504866c0ac05bdeefa53d193e3b3ef8747a9d3658fab5b8deb246400880004e9ad, salt=ac206ab5b82ad552, __v=0, sellerAddress=5eb19ff8044ab800928c613c, _id=5eb19ff8044ab800928c613b, email=seller1, updatedAt=2020-05-05T17:18:48.097Z}\u0027",
node_1           |       "caused_by": {
node_1           |         "type": "illegal_state_exception",
node_1           |         "reason": "Can\u0027t get text on a START_OBJECT at 1:146"
node_1           |       }
node_1           |     },
node_1           |     "status": 400
node_1           |   }
```

Thats the model:

```
const sellerProductSchemaDefinition: ISellerProductSchemaDefinition = {
  title: { type: String, required: true },
  category: { type: String, required: true },
  price: { type: Number, required: true },
  shortDescription: { type: String, required: true },
  longDescription: { type: String, required: true },
  seller: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Seller',
  },
  image: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'SellerProductImage',
  },
};
```

This is the typecomposer:

```

const ElasticSearchSellerProductTC = composeWithElastic({
  graphqlTypeName: 'sellerProduct',
  elasticIndex: 'sellerproduct',
  elasticType: 'sellerProduct',
  elasticMapping: {
    properties: {
      title: {
        type: 'text',
        es_indexed: true,
        es_type: 'keyword',
      },
      category: {
        type: 'text',
        es_type: 'keyword',
      },
      price: { type: 'long' },
      shortDescription: {
        type: 'text',
        es_indexed: true,
        es_type: 'keyword',
      },
      longDescription: {
        type: 'text',
        es_indexed: true,
        es_type: 'keyword',
      },
      seller: {
        type: 'string',
      },
      image: { type: 'text' },
      updatedAt: { type: 'date' },
      createdAt: { type: 'date' },
    },
  },
  elasticClient: esClient,
  // elastic mapping does not contain information about is fields are arrays or not
  // so provide this information explicitly for obtaining correct types in GraphQL
  // pluralFields: ['skills', 'languages'],
});
```

How should i cope with this?

I tried changing the mapping-type of `seller` to 

```
      seller: {
        type: 'Object',
      },
```

Which is weird anyhow, because is still says `reason": "failed to parse field [seller] of type [text]`
If i completely remove this field, everything works fine...

From my understanding it seems like the problem is that `seller` is a object, but the mapping expects a `string`...
In other words, how do i solve this problem with the relationships?

**Comment by sajadghawami** [comment]
If anyone has the same problem, this is the solution.

So i was using `mongoosastic`, and you need to transform the schema like that:

```
sellerProductSchema.plugin(mongoosastic, {
  esClient,
  index: 'sellerproduct',
  transform: (data: any, post: any) => {
    const transformedData = { ...data, seller: data.seller._id };
    return transformedData;
  },
});
```
